substrate/frame/identity/src/lib.rs

```diff
@@ -97,6 +102,9 @@ type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
 >>::NegativeImbalance;
 type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
 
+/// Maximum size of an identity we can store is 7 KiB.
+const IDENTITY_MAX_SIZE: u32 = 7 * 1024;
```

Comment:
Where does this come from and why is it private?  
is this statically enforced by testing that `Identity::max_encoded_len() <= IDENTITY_MAX_SIZE`?

Reply:
Origin [here](https://github.com/paritytech/polkadot-sdk/pull/2048#discussion_r1373512069). In short, it's the same model as in [preimage](https://github.com/paritytech/polkadot-sdk/blob/9987bbb1d264672d406071fb2e164e4e80117e13/substrate/frame/preimage/src/lib.rs#L94) but adapted to what is currently stored in this pallet.

> why is it private?

I guess it could be public. My reasoning was that users choose what `struct` to use for their `IdentityInformationProvider` and this value of the max size is unreasonably large anyway. If it didn't fail the decoding, then we'll have a hard stop in the pallet regarding the size.

> is this statically enforced by testing that `Identity::max_encoded_len() <= IDENTITY_MAX_SIZE`?

If you mean in integrity test, then not yet but it's a very good idea.

Reply:
Done, PTAL